### PR TITLE
Warp timestamp and extend max-allowable-drift for accommodate slow blocks

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Blocks have slowed, making current `MAX_ALLOWABLE_DRIFT_PERCENTAGE` again too restrictive and causing timestamp to drift away from (behind) real-world time.

#### Summary of Changes
- Remove old feature gating and deprecated code (`test_bank_hash_consistency` needs updating, since it runs without active features)
- Allow independent adjustment of max_allowable_drift values for blocks slower than PoH vs blocks faster than PoH
- Extend max_allowable_drift percentage to 80% for slow blocks: 720ms at current PoH target
- On activation, warp timestamp to current estimate with no bounding
- Also fix a i64 to u64 casting issue (unlikely to occur in the wild, since it is in the too-fast path, but feature-gated for safety first)

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
Not quite ready. One more chunk of deprecated code to remove.

**Comment by CriesofCarrots** [comment]
This is ready for a look @mvines 
The actual behavior changes (bounding adjustment, time warp) are all in https://github.com/solana-labs/solana/pull/15204/commits/1300ba4ec483f3afbbc916c9a59f0aa1b0046f81

All the other commits are cleanup/refactoring, although if you have time, I'd appreciate a sanity check on them.

File: sdk/src/feature_set.rs [hunk 1]
```diff
@@ -358,6 +347,7 @@ lazy_static! {
         (full_inflation::stakeconomy::vote::id(), "Community vote allowing Stakeconomy.com to enable full inflation"),
         (full_inflation::w3m::vote::id(), "Community vote allowing w3m to enable full inflation"),
         (full_inflation::w3m::enable::id(), "Full inflation enabled by w3m"),
+        (warp_timestamp_again::id(), "warp timestamp again, adjust bounding to 25% fast 80% slow #TODO"),
```
Comment (mvines) [inline sdk/src/feature_set.rs:350]:
```suggestion
        (warp_timestamp_again::id(), "warp timestamp again, adjust bounding to 25% fast 80% slow #15204"),
```

**Review by mvines — APPROVED** [review]

File: sdk/src/feature_set.rs (hunk 1 above)
Comment (CriesofCarrots) [reply sdk/src/feature_set.rs:350]: ty!

**Comment by mvines** [comment]
@CriesofCarrots - can this be merged?  I'm about to ship a v1.5, so we might as well get this in too

**Comment by CriesofCarrots** [comment]
> @CriesofCarrots - can this be merged? I'm about to ship a v1.5, so we might as well get this in too

Yep! Sorry for the delay

**Pull request merged**